operations/template/alert.tf

```diff
@@ -61,21 +81,70 @@ resource "azurerm_monitor_scheduled_query_rules_alert" "database_token_expired_a
     threshold = 1
   }
 
-  #   below tags are managed by CDC
   lifecycle {
     ignore_changes = [
-      tags["business_steward"],
-      tags["center"],
-      tags["environment"],
-      tags["escid"],
-      tags["funding_source"],
-      tags["pii_data"],
-      tags["security_compliance"],
-      tags["security_steward"],
-      tags["support_group"],
-      tags["system"],
-      tags["technical_steward"],
-      tags["zone"]
+      # Ignore changes to tags because the CDC sets these automagically
+      tags,
+    ]
+  }
+}
+
+resource "azurerm_monitor_metric_alert" "azure_4XX_alert" {
+  count               = local.non_pr_environment ? 1 : 0
+  name                = "cdcti-${var.environment}-azure-http-4XX-alert"
+  resource_group_name = data.azurerm_resource_group.group.name
+  scopes              = [data.azurerm_resource_group.group.id]
+  description         = "Action will be triggered when Http Status Code 4XX is greater than or equal to 1"
+  frequency           = "PT1M" // Checks every 1 minute
+  window_size         = "PT5M" // Every Check, looks back 5 minutes in history
+  //TBD: How frequent do we want this alert and how far do we want it to look back.
+
+  criteria {
+    metric_namespace = "Microsoft.Web/sites"
+    metric_name      = "Http4xx"
+    aggregation      = "Count"
+    operator         = "GreaterThanOrEqual"
+    threshold        = 1
+  }
+
+  action {
+    action_group_id = azurerm_monitor_action_group.notify_slack_email[count.index].id
+  }
+
+  lifecycle {
+    ignore_changes = [
+      # Ignore changes to tags because the CDC sets these automagically
+      tags,
+    ]
+  }
+}
+
+resource "azurerm_monitor_metric_alert" "azure_5XX_alert" {
+  count               = local.non_pr_environment ? 1 : 0
+  name                = "cdcti-${var.environment}-azure-http-5XX-alert"
+  resource_group_name = data.azurerm_resource_group.group.name
+  scopes              = [data.azurerm_resource_group.group.id]
+  description         = "Action will be triggered when Http Status Code 5XX is greater than or equal to 1"
+  frequency           = "PT1M" // Checks every 1 minute
+  window_size         = "PT5M" // Every Check, looks back 5 minutes in history
+  //TBD: How frequent do we want this alert and how far do we want it to look back.
+
+  criteria {
+    metric_namespace = "Microsoft.Web/sites"
+    metric_name      = "Http5xx"
```

Comment:
I don't think this should work since it's not a metric option but I'm unsure how to verify. Maybe use HttpServerErrors or a log alert instead if it doesn't.

Reply:
Hmmm, we could inject a crash into our code so we have a guaranteed 500.  Then deploy to the internal environment.